vm/devices/net/netvsp/src/lib.rs

```diff
@@ -1856,6 +1857,8 @@ enum PacketError {
     UnknownType(u32),
     #[error("Access")]
     Access(#[from] AccessError),
+    #[error("ExternalData")]
+    ExternalData(#[from] ExternalDataError),
```

Comment:
In general we are moving away from using `#[from]` and instead just use `#[source]`. This makes the call site more explicit so that it's clear which error variant is being constructed.

Reply:
(There are exceptions to this when it would be annoying to write `.map_err(X::Y)` a gazillion times and the inner error type is unique enough. But I don't think that applies here.)

Reply:
Fair enough. I took this feedback, and expanded the scope to make that change for existing errors in netvsp (where possible).

Reply:
Thanks for taking the feedback.

I think the `Queue` and `Access` variants in netvsp are probably common enough to count as the exceptions I was referring to. The code has become a lot less readable with the extra `.map_err` calls everywhere.

Reply:
Yeah, I'd agree. One moment.

Reply:
(there were already a number of calls to `map_err` for `WorkerError::Access`. I removed those for consistency sake, but I'm open either way here)